crates/test-programs/wasi-tests/src/utils.rs

```diff
@@ -52,3 +52,32 @@ pub unsafe fn cleanup_file(dir_fd: wasi_unstable::Fd, file_name: &str) {
 pub unsafe fn close_fd(fd: wasi_unstable::Fd) {
     assert!(wasi_unstable::fd_close(fd).is_ok(), "closing a file");
 }
+
+// Returns: (rights_base, rights_inheriting)
+pub unsafe fn fd_get_rights(fd: wasi_unstable::Fd) -> (wasi_unstable::Rights, wasi_unstable::Rights) {
```

Comment:
This function may not be necessary if we do update the test propram to use `wasi` given how it greatly simplifies things.